### PR TITLE
SDK: Implement Execution.waitToExecutionStart

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-sdk/lib/core/services/ExecutionFactory.ts [hunk 1]
```diff
@@ -24,23 +24,38 @@ export class ExecutionFactory {
...
     const waitToBeMined = totalCollected >= required ? this.createWaitToBeMined(messageHash) : async () => result.status;
-    const waitToBePending = async () => {
-      throw Error('Not implemented');
-    };
+    const waitToExecutionStart = totalCollected >= required ? this.createExecutionStart(messageHash) : async () => result.status;
     return {
       messageStatus: result.status,
       waitToBeMined,
-      waitToBePending
+      waitToExecutionStart
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/core/services/ExecutionFactory.ts]: `waitToBePending`

File: universal-login-sdk/lib/core/services/ExecutionFactory.ts [hunk 2]
```diff
@@ -24,23 +24,38 @@ export class ExecutionFactory {
...
     };
   }
 
   private isExecuted(messageStatus: MessageStatus) {
-    return !!messageStatus.transactionHash || !!messageStatus.error;
+    return messageStatus.state === 'Error' || messageStatus.state === 'Success';
+  }
+
+  private isStarted(messageStatus: MessageStatus) {
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/core/services/ExecutionFactory.ts]: hasHash

File: universal-login-sdk/lib/core/services/ExecutionFactory.ts [hunk 3]
```diff
@@ -24,23 +24,38 @@ export class ExecutionFactory {
...
+    return ['Pending', 'Success', 'Error'].includes(messageStatus.state);
+  }
+
+  private createGetStatus(messageHash: string) {
+    return async () => this.relayerApi.getStatus(messageHash);
+  }
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/core/services/ExecutionFactory.ts]: private getStatus(messageHash: string) {
    return this.relayerApi.getStatus(messageHash);
  }

File: universal-login-sdk/lib/core/services/ExecutionFactory.ts [hunk 4]
```diff
@@ -24,23 +24,38 @@ export class ExecutionFactory {
...
+
+  private createExecutionStart(messageHash: string) {
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/core/services/ExecutionFactory.ts]: createWaitForHash

File: universal-login-sdk/test/unit/core/ExecutionFactory.ts [hunk 5]
```diff
@@ -10,63 +10,95 @@ describe('UNIT: ExecutionFactory', async () => {
...
-    expect(execution.messageStatus).to.be.deep.eq(executionStatus);
-    await expect(execution.waitToBeMined()).to.be.rejectedWith('Error: waitToBeMined');
-    expect(getStatus.callCount).be.eq(callCount);
+  describe('waitToBeMined', async () => {
+    it('waitToBeMined success', async () => {
+      status.state = 'Success';
+      const execution = await executionFactory.createExecution(signedMessage);
+      const messageStatus = await execution.waitToBeMined();
+      expect(messageStatus).to.be.deep.eq(status);
+      expect(getStatus.callCount).be.eq(callCount);
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/unit/core/ExecutionFactory.ts]: expect(getStatus.callCount).be.gt(0);

File: universal-login-sdk/test/unit/core/ExecutionFactory.ts [hunk 6]
```diff
@@ -10,63 +10,95 @@ describe('UNIT: ExecutionFactory', async () => {
   let executionFactory: ExecutionFactory;
   let relayerApi: RelayerApi;
   let signedMessage: SignedMessage;
+  let defaultStatus: MessageStatus;
   let status: MessageStatus;
-  let executionStatus: MessageStatus;
-  let getStatus: SinonSpy;
+  const getStatus: SinonStub = sinon.stub().returns({status: {}});
+  const execute: SinonStub = sinon.stub();
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/unit/core/ExecutionFactory.ts]: inline

File: universal-login-sdk/test/unit/core/ExecutionFactory.ts [hunk 7]
```diff
@@ -10,63 +10,95 @@ describe('UNIT: ExecutionFactory', async () => {
...
+      expect(getStatus.callCount).be.eq(callCount);
+    });
+
+    it('error', async () => {
+      status.state = 'Error';
+      status.error = 'Error: waitToBeMined';
+
+      const execution = await executionFactory.createExecution(signedMessage);
+      expect(execution.messageStatus).to.be.deep.eq(defaultStatus);
+      await expect(execution.waitToBeMined()).to.be.rejectedWith('Error: waitToBeMined');
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/unit/core/ExecutionFactory.ts]: NO error just return

**Comment by xitronix** [comment]
Rename waitToBeMined -> waitToBeSuccess will affect the Kickback and other apps (changing interface)

**Pull request merged**